tesseract_ros/tesseract_examples/tesseract_ros_examples/src/glass_up_right_ompl_example.cpp

```diff
@@ -0,0 +1,409 @@
+  std::vector<double> swp = { -0.4, 0.2762, 0.0, -1.3348, 0.0, 1.4959, 0.0 };
+  std::vector<double> ewp = { 0.4, 0.2762, 0.0, -1.3348, 0.0, 1.4959, 0.0 };
+
+  tesseract_->getEnvironment()->setState(kin->getJointNames(), swp);
+
+  //  plotter->plotScene();
+
+  // Set Log Level
+  util::gLogLevel = util::LevelError;
+
+  // Setup Problem
+  tesseract_motion_planners::OMPLMotionPlanner ompl_planner;
+
+  tesseract_motion_planners::OMPLPlannerConfig::Ptr ompl_config;
+  if (use_constraint_)
+  {
+#ifndef OMPL_LESS_1_4_0
+    auto ompl_cnt_config =
+        std::make_shared<tesseract_motion_planners::OMPLPlannerConstrainedConfig>(tesseract_, "manipulator");
+
+    ompl_cnt_config->start_waypoint =
+        std::make_shared<tesseract_motion_planners::JointWaypoint>(swp, kin->getJointNames());
+    ompl_cnt_config->end_waypoint =
+        std::make_shared<tesseract_motion_planners::JointWaypoint>(ewp, kin->getJointNames());
+    ompl_cnt_config->collision_safety_margin = 0.01;
+    ompl_cnt_config->planning_time = planning_time_;
+    ompl_cnt_config->max_solutions = 2;
+    ompl_cnt_config->longest_valid_segment_fraction = 0.01;
+
+    ompl_cnt_config->collision_continuous = false;
+    ompl_cnt_config->collision_check = false;
+    ompl_cnt_config->simplify = false;
+    ompl_cnt_config->n_output_states = 50;
+
+    if (use_trajopt_constraint_)
+    {
+      if (plotting_)
+        ompl_cnt_config->constraint =
+            std::make_shared<TrajOptGlassUprightConstraint>(tesseract_, kin, "manipulator", "tool0", plotter);
+      else
+        ompl_cnt_config->constraint =
+            std::make_shared<TrajOptGlassUprightConstraint>(tesseract_, kin, "manipulator", "tool0", nullptr);
+    }
+    else
+    {
+      Eigen::Vector3d normal = -1.0 * Eigen::Vector3d::UnitZ();
+      ompl_cnt_config->constraint = std::make_shared<GlassUprightConstraint>(normal, kin);
+    }
+    ompl_config = ompl_cnt_config;
+#else
+    ROS_ERROR("Constraints are not supported in OMPL versions less than 1.4.0");
+    return false;
+#endif
+  }
+  else
+  {
+    auto ompl_freespace_config =
+        std::make_shared<tesseract_motion_planners::OMPLPlannerFreespaceConfig>(tesseract_, "manipulator");
+
+    ompl_freespace_config->start_waypoint =
+        std::make_shared<tesseract_motion_planners::JointWaypoint>(swp, kin->getJointNames());
+    ompl_freespace_config->end_waypoint =
+        std::make_shared<tesseract_motion_planners::JointWaypoint>(ewp, kin->getJointNames());
+    ompl_freespace_config->collision_safety_margin = 0.01;
+    ompl_freespace_config->planning_time = planning_time_;
+    ompl_freespace_config->max_solutions = 2;
+    ompl_freespace_config->longest_valid_segment_fraction = 0.01;
+
+    ompl_freespace_config->collision_continuous = false;
+    ompl_freespace_config->collision_check = false;
+    ompl_freespace_config->simplify = false;
+    ompl_freespace_config->n_output_states = 50;
+
+    ompl_config = ompl_freespace_config;
+  }
```

Comment:
If not using the constraint this is not really glass upright is it? It's just freespace around a ball. Would this be better suited to just check in the node whether or not OMPL as a sufficiently high version and then just throw an error if not? Then we could have a separate example for OMPL planning without constraints. These examples are complicated enough as they are. Personally, I think shorter, simpler, and less comprehensive examples might be more useful to people.

I don't feel strongly either way, but we should probably at least put a comment here since the results may not be the expected behavior.

Reply:
I will split into two different unit tests. I was using this for debug.

Reply:
The latest commit splits it into two examples.